stm32:  tests/samples :  add support for nucleo_h753zi  on some tests and samples

djiatsaf-st opened the pull request:
This PR adds support for the NUCLEO_H753ZI on several tests and samples for test bench purposes.
It will eventually replace the NUCLEO_H743ZI, which is obsolete, as shown here  [here](https://www.st.com/en/evaluation-tools/nucleo-h743zi.html)

djiatsaf-st committed:
samples: drivers: adc: adc_dt: allow nucleo_h753zi to run in CI

enable sample to run in CI on nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
samples: drivers: watchdog: allow nucleo_h753zi to run in CI

- enable sample to run in CI on nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
samples: net: allow nucleo_h753zi to run in CI

enable sample to run in CI on nucleo_h753zi.

related sample: ptp and zperf

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: adc: adc_api: allow nucleo_h753zi to run in CI

- add .conf and .overlay files with appropriate configurations
for nucleo_h753zi.
- enable sample to run in CI on nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: stm32_clock_configuration: allow nucleo_h753zi to run in CI

enable sample to run in CI on nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: counter: counter_basic_api: boards: add nucleo_h753zi board

Add various timer nodes to enable the test to run
successfully.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: dac: dac_api: add support for nucleo_h753zi

Add the CONFIG_BOARD_NUCLEO_H753ZI symbol to enable running
tests on the NUCLEO_H753ZI board.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
boards: st: nucleo_h753zi: update nucleo_h753zi.yaml

Add support for the RTC, WATCHDOG, and DMA peripherals.
These are required for the DMA and RTC, WATCHDOG test drivers.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: dma: chan_blen_transfer: add support for nucleo_h753zi

add .conf and .overlay files with appropriate configurations for
nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: dma: loop_transfer: add support for nucleo_h753zi

add .conf and .overlay files with appropriate configurations for
nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: flash: stm32: update testcase.yaml

Add nucleo_h753zi as an allowed platform to run this
test in CI.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: i2c_target_api: boards: add support for nucleo_h753zi

- enable i2c2 node in dts
- add .conf and .overlay with appropriate configurations for
  nucleo_h753zi.
- update testcase.yaml file for CI testing

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: pwm: pwm_api: add support for nucleo_h753zi

add .overlay file with timers1 node configuration.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: pwm: pwm_loopback: add support for nucleo_h753zi

add overlay with appropriate configuration for nucleo_h753zi

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: rtc: rtc_api: add support for nucleo_h753zi

add .conf and .overlay with appropriate configurations for
nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: uart: uart_async_api: add support for nucleo_h753zi

- add .conf and .overlay with appropriate configurations for
      nucleo_h753zi
- update testcase.yaml file for CI testing

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st commented on samples/drivers/adc/adc_dt/sample.yaml:
```diff
@@ -7,6 +7,7 @@ tests:
     depends_on: adc
     platform_allow:
       - nucleo_l073rz
+      - nucleo_h753zi
```
Yes, I understand what you mean, but I think it is valid for `integration_platforms`, not `platform_allow`. Without specifying this board, Twister will not run this test on the NUCLEO_H753ZI.

erwango reviewed (changes requested):
I have 2 main points:

- The easy one: Let's provide always the same and complete header file in overlays (Copyright plus license)
- The one to discuss: Should we remove `nucleo_h743zi` fully from the `platform_allow` ? From the `integration_platform` ?

djiatsaf-st commented:
> * The one to discuss: Should we remove `nucleo_h743zi` fully from the `platform_allow` ? From the `integration_platform` ?

- `integration_platform` :  For sure, I think we should fully remove `nucleo_h743zi` to avoid wasted resources and add `nucleo_h753zi` to reflect actively maintained boards.

- `platform_allow` : I think it is a bit more delicate. In the short term, we can keep nucleo_h743zi to maintain backward compatibility for users who may still have it. Then, we can add a comment in the board documentation (https://docs.zephyrproject.org/latest/boards/st/nucleo_h743zi/doc/index.html) or another appropriate place to suggest users switch to `nucleo_h753zi` for testing purposes. By the end of this year or later, we can fully remove `nucleo_h743zi` under platform_allow. I think it is safe to proceed this way and potentially avoid any unexpected situations.

djiatsaf-st committed:
boards: st: nucleo_h743zi: doc: add notice about board obsolescence

Important note to users about the obsolescence of NUCLEO_H743ZI.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: add nucleo_h753zi as integration platform

remove nucleo_h743zi from integration_platform and replace
with nucleo_h753zi.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st commented on tests/drivers/uart/uart_async_api/testcase.yaml:
```diff
@@ -35,7 +35,9 @@ tests:
+      - nucleo_h743zi
+      - nucleo_h753zi
     integration_platforms:
       - nucleo_h743zi
```
Removed

djiatsaf-st committed:
tests: drivers: uart: uart_basic_api: add support for nucleo_h753zi

- add .overlay with appropriate configurations for
  nucleo_h753zi
- update testcase.yaml file for CI testing

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

etienne-lms commented on tests/drivers/pwm/pwm_loopback/boards/nucleo_h753zi.overlay:
```diff
@@ -0,0 +1,40 @@
+/* 32-Bit timers */
+&timers2 {
+	status = "okay";
+	pwm2: pwm {
```
ping

etienne-lms reviewed (commented):
Commit "boards: st: nucleo_h753zi: update nucleo_h753zi.yaml".
I think this commit should come last in the series, once all required support for tests and samples are in place.
CI complains on **tests/drivers/watchdog/wdt_basic_api/** failing to build for *nucleo_h753i*. It would require a `watchdog0` DT node alias or like.

djiatsaf-st committed:
tests: drivers: watchdog: wdt_basic_api: add support for nucleo_h753zi

- add watchdog0 node alias
- remove redundant nucleo_h753zi

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>
